pkg/registry/populator_test.go

```diff
@@ -631,6 +631,37 @@ func TestListBundles(t *testing.T) {
 	}
 	require.Equal(t, 10, len(bundles))
 	require.ElementsMatch(t, expectedDependencies, dependencies)
+
+	db, cleanup = CreateTestDb(t)
+	defer cleanup()
+
+	loader, err := sqlite.NewSQLLiteLoader(db)
+	require.NoError(t, err)
+	require.NoError(t, loader.Migrate(context.TODO()))
+
+	graphLoader, err := sqlite.NewSQLGraphLoaderFromDB(db)
+	require.NoError(t, err)
+
+	query := sqlite.NewSQLLiteQuerierFromDb(db)
+
+	populate := func(bundles map[image.Reference]string) error {
+		return registry.NewDirectoryPopulator(
+			loader,
+			graphLoader,
+			query,
+			bundles,
+			make(map[string]map[image.Reference]string),
+			false).Populate(registry.ReplacesMode)
+	}
+	add := map[image.Reference]string{
+		image.SimpleReference("quay.io/test/etcd.0.9.2"):        "../../bundles/etcd.0.9.2",
+		image.SimpleReference("quay.io/test/prometheus.0.22.2"): "../../bundles/prometheus.0.22.2",
+	}
+	expectedErr := errors.NewAggregate([]error{
+		fmt.Errorf("Invalid bundle %s, bundle specifies a non-existent replacement %s", "etcdoperator.v0.9.2", "etcdoperator.v0.9.0"),
+		fmt.Errorf("Invalid bundle %s, bundle specifies a non-existent replacement %s", "prometheusoperator.0.22.2", "prometheusoperator.0.15.0"),
+	})
+	require.ElementsMatch(t, expectedErr, populate(add))
```

Comment:
I don't think `TestListBundles` is the appropriate spec for this. It probably needs its own.